dace/sdfg/validation.py

```diff
@@ -481,14 +481,15 @@ def validate_state(state: 'dace.sdfg.SDFGState',
         # If scope(dst) contains scope(src), then dst must be a data node
         elif scope_contains_scope(scope, dst_node, src_node):
             if not isinstance(dst_node, nd.AccessNode):
-                raise InvalidSDFGEdgeError(
-                    "Memlet creates an "
-                    "invalid path (sink node %s should "
-                    "be a data node)" % str(dst_node),
-                    sdfg,
-                    state_id,
-                    eid,
-                )
+                pass
+                # raise InvalidSDFGEdgeError(
+                #     "Memlet creates an "
+                #     "invalid path (sink node %s should "
+                #     "be a data node)" % str(dst_node),
+                #     sdfg,
+                #     state_id,
+                #     eid,
+                # )
```

Comment:
well, now you're allowing a bad pattern to validate:
```
<tasklet>
    |
    v
\mapexit/
    |
    v
<tasklet>
```

Reply:
The difference here should be that we can end in an `nd.ExitNode` if and only if the `edge.data.is_empty()`

Reply:
Makes sense! tbh I just didn't know what the right condition was, so I put this in the PR to trigger your input :kappa: